src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Intermediate/ClassDeclarationIntermediateNode.cs

```diff
@@ -17,7 +17,7 @@ public sealed class ClassDeclarationIntermediateNode : MemberDeclarationIntermed
 
     public string ClassName { get; set; }
 
-    public IntermediateToken BaseType { get; set; }
+    public IntermediateNode BaseType { get; set; }
```

Comment:
Consider documenting what this can be:
- BaseTypeIntermediateNode for MVC
- IntermediateToken otherwise